### ISSUE TITLE
Incorrect datetime value for available_at column

### ISSUE THREAD

**Issue by ammezie** [cavai-research/Adonis-Queue#29]
I'm getting the error before when a try to queue a job. I'm using MySQL and I think it has to do with the way MySQL/PostgreSQL handles datetime.

```
[10:38:11.921] FATAL (adonis-background-job-demo/2916): "uncaughtException" detected
    Error: insert into `adonis_jobs` (`available_at`, `class_path`, `payload`) values ('2023-07-09 10:38:11.915 +01:00', 'app/Jobs/ExampleJob.ts', '{\"json\":{\"classPath\":\"app/Jobs/ExampleJob.ts\",\"data\":[{\"user\":{\"name\":\"John\",\"email\":\"john@test.com\",\"created_at\":\"2023-07-09T10:38:11.726+01:00\",\"updated_at\":\"2023-07-09T10:38:11.726+01:00\",\"id\":1},\"signedUrl\":\"/verification/john@test.com?signature=eyJtZXNzYWdlIjoiL3ZlcmlmaWNhdGlvbi9qb2huQHRlc3QuY29tIiwicHVycG9zZSI6ImVtYWlsIHZlcmlmaWNhdGlvbiJ9.-UyIPSjUUn1pWgBZiiEDVKjZ9gbxGsK1_BrkxCYs50g\"}],\"version\":\"v1\"}}') - Incorrect datetime value: '2023-07-09 10:38:11.915 +01:00' for column 'available_at' at row 1
        at Packet.asError (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/packets/packet.js:728:17)
        at Query.execute (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/commands/command.js:29:26)
        at Connection.handlePacket (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/connection.js:497:34)
        at PacketParser.onPacket (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/connection.js:97:12)
        at PacketParser.executeStart (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/packet_parser.js:75:16)
        at Socket.<anonymous> (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/connection.js:104:25)
        at Socket.emit (node:events:527:28)
        at addChunk (node:internal/streams/readable:315:12)
        at readableAddChunk (node:internal/streams/readable:289:9)
        at Socket.Readable.push (node:internal/streams/readable:228:10)
[10:38:11.954] WARN (adonis-background-job-demo/2916): pino.final with prettyPrint does not support flushing

/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/packets/packet.js:728
    const err = new Error(message);
                ^
Error: insert into `adonis_jobs` (`available_at`, `class_path`, `payload`) values ('2023-07-09 10:38:11.915 +01:00', 'app/Jobs/ExampleJob.ts', '{\"json\":{\"classPath\":\"app/Jobs/ExampleJob.ts\",\"data\":[{\"user\":{\"name\":\"John\",\"email\":\"john@test.com\",\"created_at\":\"2023-07-09T10:38:11.726+01:00\",\"updated_at\":\"2023-07-09T10:38:11.726+01:00\",\"id\":1},\"signedUrl\":\"/verification/john@test.com?signature=eyJtZXNzYWdlIjoiL3ZlcmlmaWNhdGlvbi9qb2huQHRlc3QuY29tIiwicHVycG9zZSI6ImVtYWlsIHZlcmlmaWNhdGlvbiJ9.-UyIPSjUUn1pWgBZiiEDVKjZ9gbxGsK1_BrkxCYs50g\"}],\"version\":\"v1\"}}') - Incorrect datetime value: '2023-07-09 10:38:11.915 +01:00' for column 'available_at' at row 1
    at Packet.asError (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/packets/packet.js:728:17)
    at Query.execute (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/commands/command.js:29:26)
    at Connection.handlePacket (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/connection.js:497:34)
    at PacketParser.onPacket (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/connection.js:97:12)
    at PacketParser.executeStart (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/packet_parser.js:75:16)
    at Socket.<anonymous> (/Users/~/Workspace/adonis-background-job-demo/node_modules/mysql2/lib/connection.js:104:25)
    at Socket.emit (node:events:527:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
[ warn ]  Underlying HTTP server died with "1 code"
```

**Comment by ndianabasi** [comment]
Referring to the segment: https://github.com/cavai-research/Adonis-Queue/blob/e9216df2cff3298888d2285248d06d5fabbd6175/src/Drivers/Database.ts#L15-L19

`availableAt` is directly assigned to the `insert` method without any further parsing.

If you are using `Luxon DateTime`, call `toString()` or `toSQL()` on the DateTime instance before assigning to `availableAt`.

If you are using native `Date` object, you can pass in the Date object directly or call `toString()` on the Date instance.

As an improve, I think the code can be improved so that `availableAt` expects a Luxon instance. cc @McSneaky

**Comment by ammezie** [comment]
I think the `available_at` column is supposed to be set automatically at the point of calling `dispatch()`.

From the look of things, `availableAt` is already a type of luxon https://github.com/cavai-research/Adonis-Queue/blob/e9216df2cff3298888d2285248d06d5fabbd6175/src/types.ts#L14

**Comment by McSneaky** [comment]
@ammezie this should be fixed in 2.0.4 https://github.com/cavai-research/Adonis-Queue/releases/tag/v2.0.4
